Mathlib/FieldTheory/Finite/Extension.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Kevin Buzzard
+-/
+import Mathlib.FieldTheory.Finite.GaloisField
+
+/-!
+# Extensions of finite fields
+
+In this file we develop the theory of extensions of finite fields.
+
+If `k` is a finite field (of cardinality `q = p ^ m`), then there is a unique (up to in general
+non-unique isomorphism) extension `l` of `k` of any given degree `n > 0`.
+
+This extension is Galois with cyclic Galois group of degree `n`, and the (arithmetic) Frobenius map
+`x ↦ x ^ q` is a generator.
+
+
+## Main definition
+
+* `FiniteField.Extension k p n` is a non-canonically chosen extension of `k` of degree `n`
+  (for `n > 0`).
+
+## Main Results
+
+* `FiniteField.algEquivExtension`: any other field extension `l/k` of degree `n` is (non-uniquely)
+isomorphic to our chosen `FiniteField.Extension k p n`.
+
+-/
+
+variable (k : Type*) [Field k] [Finite k]
+variable (p : ℕ) [Fact p.Prime] [Algebra (ZMod p) k]
```

Comment:
From a usability standpoint, do we really want to fix a `Algebra (ZMod p) k`?
I think we should instead do `CharP k p` and 
```
attribute [local instance] ZMod.algebra k p in
def Extension := GaloisField p (Module.finrank (ZMod p) k * n)
```
and then only assume `Algebra (ZMod p) k` when statements need them.